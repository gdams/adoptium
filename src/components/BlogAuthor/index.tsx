import React from 'react';
import { FaGithub, FaTwitter, FaLinkedin } from 'react-icons/fa';
import { Link } from 'gatsby';

import ProfilePic from '../ProfilePic';

const GitHubLink = (props) => {
  if (!props.name) {
    return null;
  }

  return (
    <a className='no-underline' aria-label='GitHub Profile' href={`https://github.com/${props.name}`}><FaGithub /></a>
  );
};

const TwitterLink = (props) => {
  if (!props.name) {
    return null;
  }

  return (
    <a className='no-underline' aria-label='Twitter Profile' href={`https://twitter.com/${props.name}`}><FaTwitter /></a>
  );
};

const LinkedinLink = (props) => {
  if (!props.name) {
    return null;
  }

  return (
    <a className='no-underline' aria-label='LinkedIn Profile' href={`https://www.linkedin.com/in/${props.name}`}><FaLinkedin /></a>
  );
};

const BlogAuthor = (props) => {
  const author = props.author;
  const identifier = props.identifier;
  const href = `/blog/author/${identifier}`;

  return (
    <div
      style={{
        display: 'flex',
      }}
    >
      <ProfilePic identifier={identifier} name={author.name} />
      <p>
        Posted by <Link to={href}>{author.name}</Link>
        {author.summary &&
         <>â€“ {author.summary}</>
        }
        {' '}
        <GitHubLink name={author.github} />
        {' '}
        <TwitterLink name={author.twitter} />
        {' '}
        <LinkedinLink name={author.linkedin} />
      </p>
    </div>
  );
};

export default BlogAuthor;
