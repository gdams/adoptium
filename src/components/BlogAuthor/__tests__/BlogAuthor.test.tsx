import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import BlogAuthor from '../index';

describe('BlogAuthor component', () => {
    it('should render correctly', () => {
        const { container } = render(
            <BlogAuthor
                author='Adoptium PMC'
                date='2021-07-01'
                identifier='pmc'
            />
        );

        expect(container).toMatchSnapshot();
    });
});