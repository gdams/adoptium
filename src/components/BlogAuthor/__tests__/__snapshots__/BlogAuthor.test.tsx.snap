// Vitest Snapshot v1

exports[`BlogAuthor component > should render correctly 1`] = `
<div>
  <div
    style="display: flex;"
  >
    <img
      image="[object Object]"
      stlye="[object Object]"
      style="margin-right: 0.5rem; margin-bottom: 0px; min-width: 50px; border-radius: 100%;"
    />
    <p>
      Posted by 
      <a
        href="/blog/author/pmc"
      />
       
       
       
    </p>
  </div>
</div>
`;
